packages/override-tools/src/CrossProcessLock.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ *
+ * @format
+ */
+
+import * as net from 'net';
+
+/**
+ * CrossProcessLock implements a non-polling cross-process asynchronus barrier.
+ *
+ * - The lock is acquired by creating a named pipe
+ *
+ * - If named pipe creation fails due to an existing pipe, the process
+ *   subscribes to the pipe, waiting asynchronusly until it dies.
+ *
+ * - Once the pipe dies, any subscribers will race to acquire the lock (create
+ *   the pipe server)
+ *
+ */
+export default class CrossProcessLock {
```

Comment:
>CrossProcessLock [](start = 21, length = 16)

Doesn't javascript already offer some library like this?

Reply:
I did find some, but they were unmaintained and had pretty questionable implementations. Likely didn't have Typescript bindings either.